scripts/get-external-data.py

```diff
@@ -310,73 +333,74 @@ def main():
-                        zip = zipfile.ZipFile(io.BytesIO(download.content))
-                        for member in source["archive"]["files"]:
-                            zip.extract(member, workingdir)
-
-                    ogrpg = "PG:dbname={}".format(database)
-
-                    if port is not None:
-                        ogrpg = ogrpg + " port={}".format(port)
-                    if user is not None:
-                        ogrpg = ogrpg + " user={}".format(user)
-                    if host is not None:
-                        ogrpg = ogrpg + " host={}".format(host)
-                    if password is not None:
-                        ogrpg = ogrpg + " password={}".format(password)
-
-                    ogrcommand = ["ogr2ogr",
-                                  '-f', 'PostgreSQL',
-                                  '-lco', 'GEOMETRY_NAME=way',
-                                  '-lco', 'SPATIAL_INDEX=FALSE',
-                                  '-lco', 'EXTRACT_SCHEMA_FROM_LAYER_NAME=YES',
-                                  '-nln', "{}.{}".format(config["settings"]["temp_schema"], name)]
-
-                    if "ogropts" in source:
-                        ogrcommand += source["ogropts"]
-
-                    ogrcommand += [ogrpg,
-                                   os.path.join(workingdir, source["file"])]
-
-                    logging.info("  Importing into database")
-                    logging.debug("running {}".format(
-                        subprocess.list2cmdline(ogrcommand)))
-
-                    # ogr2ogr can raise errors here, so they need to be caught
-                    try:
-                        subprocess.check_output(
-                            ogrcommand, stderr=subprocess.PIPE, universal_newlines=True)
-                    except subprocess.CalledProcessError as e:
-                        # Add more detail on stdout for the logs
-                        logging.critical(
-                            "ogr2ogr returned {} with layer {}".format(e.returncode, name))
-                        logging.critical("Command line was {}".format(
-                            subprocess.list2cmdline(e.cmd)))
-                        logging.critical("Output was\n{}".format(e.output))
-                        logging.critical("Error was\n{}".format(e.stderr))
-                        raise RuntimeError(
-                            "ogr2ogr error when loading table {}".format(name))
-
-                    logging.info("  Import complete")
-
-                    this_table.index()
-                    if renderuser is not None:
-                        this_table.grant_access(renderuser)
-                    this_table.replace(download.last_modified)
+                    continue
+
+
+                shutil.rmtree(workingdir, ignore_errors=True)
+                os.makedirs(workingdir, exist_ok=True)
+                if "archive" in source and source["archive"]["format"] == "zip":
```

Comment:
From here on, changes are mostly identation, I don't know how to make github not display those, but `git diff` has `--ignore-all-space`

Reply:
Aha, I found how to do it!